### PR TITLE
Created mocked login feature with clean architecture and created tests

### PR DESCRIPTION
**Pull request by roandersonpinheiro**

### CODE REVIEW

File: android-client/app/src/main/java/com/example/habits/data/repository/LoginRepositoryImpl.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+class LoginRepositoryImpl(
+    private val service: UserService
+) : LoginRepository {
+
+    override suspend fun login(login: String): Flow<Resource<User>> = flow {
+        try {
+            val resultLogin = service.login(login)
+            emit(Resource.Success(data = resultLogin.toModel()))
+        } catch (ex: HttpException) {
+            val error = RemoteException("Unable to retrieve posts")
```
Comment (edimossilva) [inline android-client/app/src/main/java/com/example/habits/data/repository/LoginRepositoryImpl.kt:20]: "Unable to retrieve posts"?

File: android-client/app/src/main/java/com/example/habits/data/services/UserService.kt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.example.habits.data.services
+
+import com.example.habits.data.entities.network.UserDTO
+import retrofit2.http.Body
+import retrofit2.http.POST
+
+interface UserService {
+    @POST("users/sign_in")
```
Comment (edimossilva) [inline android-client/app/src/main/java/com/example/habits/data/services/UserService.kt:8]: have you tried it on with [fake backend server](https://github.com/espoo-dev/group-habits/tree/main/fake-json-server)?

File: android-client/app/src/test/java/com/example/habits/MainDIComponentTest.kt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package com.example.habits
+
+import org.koin.core.context.loadKoinModules
+import org.koin.core.context.startKoin
+
+const val BASE_URL = "localhost:3000/api/v1/"
```
Comment (edimossilva) [inline android-client/app/src/test/java/com/example/habits/MainDIComponentTest.kt:6]: should this be an environment variable or some sort of config var?

File: android-client/app/src/test/java/com/example/habits/data/repository/UserRepositoryTest.kt [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    private val loginRepository = mockk<LoginRepository>()
+
+    @Test(expected = RemoteException::class)
+    fun `should return exception with error`() = runBlocking {
+
+        // GIVEN
+        coEvery { loginRepository.login("") } throws RemoteException("")
+
+        // WHEN
+        val result = loginRepository.login("").first()
```
Comment (edimossilva) [inline android-client/app/src/test/java/com/example/habits/data/repository/UserRepositoryTest.kt:28]: this var is not being used, should we remove it?

File: android-client/app/src/test/java/com/example/habits/domain/LoginUseCaseTest.kt [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    @Test(expected = RemoteException::class)
+    fun `should return exception with error`() = runBlocking {
+
+        // GIVEN
+        coEvery { repository.login("") } throws RemoteException("")
+
+        // WHEN
+        val result = login.execute().first()
```
Comment (edimossilva) [inline android-client/app/src/test/java/com/example/habits/domain/LoginUseCaseTest.kt:46]: should we remove this?

**Review by edimossilva — APPROVED** [review]
LGTM!
I've done some small comments, nothing relevant.

I am also down to renaming the package `com.example.habits` to `com.espoodev.habits`

File: android-client/app/src/main/java/com/example/habits/data/services/UserService.kt (hunk 2 above)
Comment (roandersonpinheiro) [reply android-client/app/src/main/java/com/example/habits/data/services/UserService.kt:8]: I mocked the user object in tests

File: android-client/app/src/test/java/com/example/habits/domain/LoginUseCaseTest.kt (hunk 5 above)
Comment (roandersonpinheiro) [reply android-client/app/src/test/java/com/example/habits/domain/LoginUseCaseTest.kt:46]: done

File: android-client/app/src/test/java/com/example/habits/MainDIComponentTest.kt (hunk 3 above)
Comment (roandersonpinheiro) [reply android-client/app/src/test/java/com/example/habits/MainDIComponentTest.kt:6]: configuration variable

File: android-client/app/src/main/java/com/example/habits/data/repository/LoginRepositoryImpl.kt (hunk 1 above)
Comment (roandersonpinheiro) [reply android-client/app/src/main/java/com/example/habits/data/repository/LoginRepositoryImpl.kt]: Done

**Pull request merged**